scipy/linalg/basic.py

```diff
@@ -1125,3 +1125,146 @@ def pinvh(a, cond=None, rcond=None, lower=True, return_rank=False,
         return B, len(psigma_diag)
     else:
         return B
+
+def balance(A, permute=True, scale=True, separate = False,
+                 silent=False, overwrite_a=False, scaled_block=False):
+    """
+    A wrapper around LAPACK's xGEBAL routine family for matrix balancing. 
+    
+    The balancing tries to equalize the row and column 1-norms by applying 
+    a similarity transformation such that the magnitude variation of the 
+    matrix entries is reflected to the scaling matrices. This is particularly 
+    useful for eigenvalue and matrix decompositions and often automatically 
+    performed. Hence, this function provides a convenience for manually 
+    balancing. 
+        
+    Moreover, if turned on, the matrix is first permuted to isolate the upper 
+    triangular parts of the matrix and only the remaining part is scaled. 
+    
+    The balanced matrix satisfies the following equality 
+    
+    .. math::
+         
+                        B = T^{-1} A T 
+
+    The algorithm is based on the well-known technique of [1]_ and has 
+    been modified to account for special cases. See [2]_ for details
+    which have been implemented since LAPACK v3.5.0. 
+    
+    
+    Parameters
+    ----------
+    A : (n, n) array_like
+        Square data matrix for the balancing.
+        
+    permute : bool
+        The selector to define whether permutation of A is also performed
+        prior to scaling.
+    
+    scale : bool
+        The selector to turn on and off the scaling. If False, the matrix 
+        will not be scaled. 
+        
+    separate : bool
```

Comment:
Aha! That's a very good point. I totally forgot about this annoyance and still coded this way. Sorry for that. 

What is your thought about the following situation? 

1) info_code is redundant because LAPACK can't trip silently since it passes through my hoops and receives either valid data or I raise an exception. Otherwise there are bigger problems that is beyond this function's authority :) 

2) Depending on the separate boolean value I either return `T` as ndarray or a tuple holding `p` and `s` 1D arrays. 

3) scaled_blocks is always turned on and the boolean is removed from the arguments. 

Finally, thanks for the review. I'll check the remaining parts now.